### PR TITLE
Continue Looking for Route Handler until match handles it (#131)

### PR DESCRIPTION
**Pull request by glentregoning**

### CODE REVIEW

**Comment by glentregoning** [comment]
[clabot:check]

File: Tests/Router/DPLDeepLinkRouterSpec.m [hunk 1]
```diff
@@ -139,47 +140,41 @@
...
                 expect(handled).to.beTruthy();
                 expect(error).to.beNil();
                 done();
             }];
+            expect(didMatchOnSecondRegistration).to.beTruthy();
             expect(isHandled).to.beTruthy();
         });
     });
     
-    it(@"produces an error when a URL has no matching route", ^{
```
Comment (wessmith) [inline Tests/Router/DPLDeepLinkRouterSpec.m]: Why has this test been removed? Certainly we should still test that if there are no handlers we produce an error.

File: Tests/Router/DPLDeepLinkRouterSpec.m [hunk 2]
```diff
@@ -139,47 +140,41 @@
...
-            BOOL isHandled = [router handleURL:url withCompletion:^(BOOL handled, NSError *error) {
-                expect(handled).to.beFalsy();
-                expect(error.code).to.equal(DPLRouteNotFoundError);
-                done();
-            }];
-            expect(isHandled).to.beFalsy();
-        });
-    });
-    
-    it(@"produces an error when a route handler does not specify a target view controller", ^{
```
Comment (wessmith) [inline Tests/Router/DPLDeepLinkRouterSpec.m]: Looks like this test was removed too. It's still invalid to not specify a target view controller. Why is this test removed?

File: Tests/Router/DPLDeepLinkRouterSpec.m (hunk 1 above)
Comment (glentregoning) [reply Tests/Router/DPLDeepLinkRouterSpec.m]: Hrm maybe this was added since this PR was prepared? I don’t remember removing it. But I can add it back

**Comment by wessmith** [comment]
Merged in #154

**Pull request closed without merging**